### ISSUE TITLE
Pytorch中的Dataset, Sampler, Dataloader

### ISSUE THREAD

**Issue by dagrons** [dagrons/blog#4]
## 总结

Dataset形成下标和(data, label)的映射关系，Dataset的__getitem__(self, index)每次调用返回一个(data, label)

Sampler每次返回一个index_list

Dataloader接受一个Dataset和Sampler, 工作时，每次从Sampler中获取index_list，然后通过index_list到Dataset中获取对应的数据和标签

## 一些细节

Dataset需要实现__len__和__getitem__方法

Sampler需要实现__len__和__iter__方法

## 大致如下

![image](https://user-images.githubusercontent.com/61461791/136657555-479ef70c-dbec-449f-986d-bad277fb1a86.png)
